hack/lib/golang.sh

```diff
@@ -206,25 +208,29 @@ kube::golang::setup_platforms() {
     readonly KUBE_CLIENT_PLATFORMS
 
   elif [[ "${KUBE_FASTBUILD:-}" == "true" ]]; then
-    KUBE_SERVER_PLATFORMS=(linux/amd64)
+    host_arch=$(kube::util::host_arch)
+    if [[ "${host_arch}" != "amd64" && "${host_arch}" != "arm64" ]]; then
```

Comment:
can we add a comment justifying this?

Reply:
Done! thanks.

Reply:
er, but why not do this for all platforms? I think that's still missing.

Reply:
Ping 🙃

Reply:
@BenTheElder limited by `kube-cross`, let's roll this out for arm64 (linux + darwin) and then we can hit the other arch(es)

Reply:
ACK.

Reply:
we should probably also update the makefile docs / help text, but that can be a follow up